src/index.js

```diff
@@ -120,22 +138,39 @@ async function handleDelete({ config, provider }, change, log) {
 }
 
 /**
- * Handle a single update for an index provider.
+ * Handle a single update for an index handler.
  *
  * @param {object} param0 parameters
  * @param {Change} change change
  * @param {object} log logger
  */
 async function handleUpdate({
-  config, provider, path, body,
+  config, handler, path, include, body,
 }, change, log) {
-  if (!provider) {
+  if (!handler) {
     return {
       status: 400,
-      reason: 'Provider not available, parameters missing or target unsuitable',
+      reason: 'Handler not available, parameters missing or target unsuitable',
     };
   }
-  if (!path) {
+  if (!include) {
+    if (change.uid) {
+      // This could be a move from our input domain to some region outside, so verify
+      // we do not keep a record in the index for an item we no longer track
+      try {
+        const result = await handler.delete({ sourceHash: change.uid });
+        if (result.status !== 404) {
+          // yes, the item was present in our index, so return that result
+          return result;
+        }
+      } catch (e) {
+        log.error(`An error occurred deleting record ${change.uid} in ${config.name}`, e);
+        return {
+          status: 500,
+          reason: e.message,
+        };
+      }
+    }
```

Comment:
This new section covers the "unpublish" use case, when an item is moved outside.